File: src/main/java/javasst/scanner/JavaSstToken.java
package javasst.scanner;

import javasst.JavaSstType;
import scanner.Token;

import java.io.File;

/**
 * This class contains constants for all terminal symbols.
 */
public class JavaSstToken extends Token<JavaSstType> {

    /**
     * Create a new token.
     *
     * @param identifier The identifier.
     * @param type       The type.
     * @param line       The line.
     * @param column     The column.
     * @param file       The file.
     */
    public JavaSstToken(final String identifier, final JavaSstType type, final int line, final int column, final File file) {
        super(identifier, type, line, column, file);
    }
}

File: src/main/scala/javasst/scanner/JavaSstToken.scala
package javasst.scanner

import javasst.JavaSstType.JavaSstType

import compiler.scanner.Token

/**
  * This class contains constants for all terminal symbols.
  *
  * @author Julian Schlichtholz
  * @version 1.0.0
  */
class JavaSstToken(override val identifier: String, override val typ: JavaSstType, override val line: Int,
                   override val column: Int, override val source: String)
  extends Token[JavaSstType](identifier, typ, line, column, source) {
}
